.changelog/2734.txt

```diff
@@ -0,0 +1,3 @@
+```releast-note:feature
+api-gateway: Add CRD for GatewayPolicy and HTTPRouteAuthFilter
```

Comment:
It seems like you're going to separate this into two commits back to `main`? Or were you planning on squashing this along with the actual implementation?

Reply:
I'm not sure I follow, this includes the implementation for the CRDs, but not necessarily the underlying logic for them to actually do anything, would it make sense to have this changelog on a later commit? (we would still need to handle the translation work and all that to make them usable)

Reply:
Yeah, so that was my question, are you going to separate the two:

1. CRDs
2. the plumbing to use them

I'm just wondering if it makes sense to just work off this branch and then do a monolithic feature merge -- just because I envision that the actual controller plumbing here is going to be fairly light (probably roughly the same amount as Sarah's previous [PR](https://github.com/hashicorp/consul-k8s/pull/2735) or a bit larger due to the use of both policies and filters). If you do that, I'd probably just do a single changelog entry, as this commit is pretty much unusable from a user-perspective without the actual implementation paired with it, so not sure if a changelog actually makes sense here.

Reply:
so I created a new branch that we're going to work off of that this branch is now based on